crates/red_knot_python_semantic/src/types.rs

```diff
@@ -3521,21 +3530,17 @@ impl<'db> Type<'db> {
         self,
         db: &'db dyn Db,
         name: &str,
-        mut argument_types: CallArgumentTypes<'_, 'db>,
+        argument_types: &mut CallArgumentTypes<'_, 'db>,
```

Comment:
Taking in a `mut` reference here let's us use `with_self` down below, which reuses and modifies a `CallArgumentTypes` in place.